pck/utils.go

```diff
@@ -86,31 +87,42 @@ func ArtifactFromQuery(q Query) Artifact {
 	return a
 }
 
-func GenerateReferrerTree(repo *remote.Repository, name string, artifact string) ([]Referrer, error) {
+func GenerateReferrerTreeConcurrent(repo *remote.Repository, name string, artifact string) ([]Referrer, error) {
 	ctx := context.Background()
 
 	descriptor, err := oras.Resolve(ctx, repo, name, oras.DefaultResolveOptions)
-
 	if err != nil {
 		return nil, err
 	}
 
+	var wg sync.WaitGroup
+	var mu sync.Mutex
 	result := []Referrer{}
+
 	if err := repo.Referrers(ctx, descriptor, "", func(referrers []v1.Descriptor) error {
 		for _, referrer := range referrers {
-			nodes, _ := GenerateReferrerTree(repo, artifact+ATSYMBOL+string(referrer.Digest), artifact)
-			res := Referrer{
-				Ref:   referrer,
-				Nodes: nodes,
-			}
-
-			result = append(result, res)
+			wg.Add(1)
+			go func(ref v1.Descriptor) {
+				defer wg.Done()
+
+				nodes, _ := GenerateReferrerTreeConcurrent(repo, artifact+ATSYMBOL+string(ref.Digest), artifact)
```

Comment:
The returned error should not be ignored

Reply:
I think the returned error can be ignored.
- this part of code is for subsequent recursive calls and not the first function call to GenerateReferrerTreeConcurrent.
- its just that, if there's an error in subsequent calls, then there would be no referencing artifacts, and
```golang
res := Referrer {
  Ref:   referrer,
  Nodes: nodes, // this would be empty array [ ] then
}
```
- the error in fetching first order (very first referencing artifacts of the tree) is already handled in handlers.Referrers as,
```golang
referrers, err := GenerateReferrerTreeConcurrent(repo, a.Name, artifact)

if err != nil {
```

Reply:
We can have a meeting to discuss tmr